src/basic/hash-funcs.h

```diff
@@ -16,11 +16,17 @@ struct hash_ops {
 };
 
 #define _DEFINE_HASH_OPS(uq, name, type, hash_func, compare_func, free_key_func, free_value_func, scope) \
-        _unused_ static void (* UNIQ_T(static_hash_wrapper, uq))(const type *, struct siphash *) = hash_func; \
-        _unused_ static int (* UNIQ_T(static_compare_wrapper, uq))(const type *, const type *) = compare_func; \
+        static void UNIQ_T(static_hash_wrapper, uq)(const void *p, struct siphash *state) { \
+                const type *q = p;                                      \
+                hash_func(q, state);                                    \
```

Comment:
you should be able to pass `p` here directly, no need for the indirection of `q`